### DISCUSSION TITLE
Switching between different Application Shell Layout

### DISCUSSION THREAD

**Discussion by maheshrajrp** [eclipse-theia/theia#11744] [General] [answered]
Hi,
Can we have more than one kind of application shell layout in theia that user can switch between ?

For eg.
Application Shell 1: Normal Mode -> Vanilla Theia
Application Shell 2: Debug Mode -> Different sets of custom editors, tree widgets, menu & status-bar (Entirely new sets of widgets)

In-order to Switch maybe we could have a command registered. A service to share info between the two layouts.

What would be the best way to switch between them as they have no widgets in common. Is there any theia implementation which I could refer to ?

**Comment by vince-fugnitto** [comment, marked as answer]
@maheshrajrp I believe you already asked the question [here](https://github.com/eclipse-theia/theia/discussions/11586)?

**Reply by maheshrajrp** [reply]
Yes,  but the use- case is changed abit. Since it wasn't replied I have created a new discussion.  Hope that is fine.

**Reply by vince-fugnitto** [reply]
You can always edit the previous post if something has changed, not need to duplicate :)

As for an answer, we generally try to answer if we have information. In this case I believe you'd be the first one to implement a secondary application shell so everything is fairly new and there is current implementation to do so.

You may try extending your `application-shell` implementation to not really support additional shells but perhaps support switching `ApplicationShell.LayoutData` based on your needs.

**Reply by maheshrajrp** [reply]
@vince-fugnitto Got you point.  Will keep that in mind.  Thanks for the suggestion too. Will try that out.
